plugins/module_utils/network/sonic/argspec/route_maps/route_maps.py

```diff
@@ -0,0 +1,178 @@
+                                },
+                                'vni': {'type': 'int'}
+                            },
+                            'required_one_of': [['default_route', 'route_type', 'vni']],
+                            'type': 'dict'
+                        },
+                        'ext_comm': {'type': 'str'},
+                        'interface': {'type': 'str'},
+                        'ip': {
+                            'options': {
+                                'address': {'type': 'str'},
+                                'next_hop': {'type': 'str'}
+                            },
+                            'required_one_of': [['address', 'next_hop']],
+                            'type': 'dict'
+                        },
+                        'ipv6': {
+                            'options': {
+                                'address': {
+                                    'required': True,
+                                    'type': 'str'
+                                }
+                            },
+                            'type': 'dict'
+                        },
+                        'local_preference': {'type': 'int'},
+                        'metric': {'type': 'int'},
+                        'origin': {
+                            'choices': ['egp', 'igp', 'incomplete'],
+                            'type': 'str'
+                        },
+                        'peer': {
+                            'mutually_exclusive': [['ip', 'ipv6', 'interface']],
+                            'options': {
+                                'interface': {'type': 'str'},
+                                'ip': {'type': 'str'},
+                                'ipv6': {'type': 'str'}
+                            },
+                            'required_one_of': [['ip', 'ipv6', 'interface']],
+                            'type': 'dict'
+                        },
+                        'source_protocol': {
+                            'choices': ['bgp', 'connected', 'ospf', 'static'],
+                            'type': 'str'
+                        },
+                        'source_vrf': {'type': 'str'},
+                        'tag': {'type': 'int'}
+                    },
+                    'type': 'dict'
+                },
+                'set': {
+                    'options': {
+                        'as_path_prepend': {'type': 'str'},
+                        'comm_list_delete': {'type': 'str'},
+                        'community': {
+                            'options': {
+                                'community_number': {
+                                    'elements': 'str',
+                                    'type': 'list'
+                                },
+                                'community_attributes': {
+                                    'elements': 'str',
+                                    'type': 'list'
```

Comment:
Thanks for pointing this out. I have only seen "choices" used for dict items and didn't realize that it could be applied to list items. The fact that the argspec definition of a list is specified with a dict does make this possible, so I should have researched it. I appreciate your doing that and relaying the availability of this option.

I have implemented this suggestion.